packages/ckeditor5-ckbox/src/ckboxcommand.ts

```diff
@@ -299,6 +303,25 @@ export default class CKBoxCommand extends Command {
 		if ( selection.isCollapsed ) {
 			const selectionAttributes = toMap( selection.getAttributes() );
 			const textNode = writer.createText( linkName, selectionAttributes );
+
+			if ( writer && assetsCount > 1 ) {
```

Comment:
When can `writer` be `null`? I'm surprised there is 100% for this PR.

Reply:
`writer` is passed only when asset is not an `image` :point\_right: https://github.com/ckeditor/ckeditor5/blob/3cbdb2e85e92b4392d70e1943a3efab72591e7b7/packages/ckeditor5-ckbox/src/ckboxcommand.ts#L261

Reply:
`writer` is passed only when asset is not an `image` :point\_right: https://github.com/ckeditor/ckeditor5/blob/3cbdb2e85e92b4392d70e1943a3efab72591e7b7/packages/ckeditor5-ckbox/src/ckboxcommand.ts#L261

Reply:
In the `if` statement you linked, the `_insertImage` method is called if the asset is an image. Otherwise, the `_insertLink` method is called. These method have different signatures - the latter method accepts the `writer` object, while the former does not.

In both `_insertAsset` and `_insertLink`, the `writer` type indicates that it's a required parameter. If that's really the case, then there's no need to check if `writer` exists. Otherwise, the type of the `writer` should be updated to `writer?: Writer`.

Reply:
> In the `if` statement you linked, the `_insertImage` method is called if the asset is an image. Otherwise, the `_insertLink` method is called. These method have different signatures - the latter method accepts the `writer` object, while the former does not.

Yes, you have right! My bad, didn't spot it that there are called different functions :see\_no\_evil: